packages/scripts/src/addOrbitChain/schemas.ts

```diff
@@ -0,0 +1,317 @@
+      chain.tokenBridge.parentWeth,
+      chain.tokenBridge.parentWethGateway,
+    ].filter(
+      (address): address is string =>
+        typeof address === "string" &&
+        address !== "0x0000000000000000000000000000000000000000"
+    );
+
+    const childAddressesToCheck = [
+      chain.tokenBridge.childCustomGateway,
+      chain.tokenBridge.childErc20Gateway,
+      chain.tokenBridge.childGatewayRouter,
+      chain.tokenBridge.childMulticall,
+      chain.tokenBridge.childProxyAdmin,
+    ].filter(
+      (address): address is string =>
+        typeof address === "string" &&
+        address !== "0x0000000000000000000000000000000000000000"
+    );
+
+    const checkAddresses = async (
+      addresses: string[],
+      rpcUrl: string,
+      blockExplorer: string,
+      chainId: number,
+      chainName: string
+    ) => {
+      const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+      for (const address of addresses) {
+        try {
+          const code = await provider.getCode(address);
+          if (code === "0x") {
+            const explorerLink = `${blockExplorer}/address/${address}`;
+            console.warn(
+              `Address ${address} on ${chainName} (chainId: ${chainId}) is not a contract. Verify manually: ${explorerLink}`
+            );
+            // TODO: Uncomment this when we can verify all contracts
+            // ctx.addIssue({
+            //   code: z.ZodIssueCode.custom,
+            //   message: `Address at ${address} is not a contract on ${chainName}. Verify manually: ${explorerLink}`,
+            // });
+          }
+        } catch (error) {
+          const explorerLink = `${blockExplorer}/address/${address}`;
+          console.log(
+            `Error checking contract at ${address} on ${chainName} (chainId: ${chainId}). Verify manually: ${explorerLink}`
+          );
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Error checking contract at ${address} on ${chainName}. Verify manually: ${explorerLink}`,
+          });
+        }
+      }
+    };
+
+    await checkAddresses(
+      parentAddressesToCheck,
+      parentChainInfo.rpcUrl,
+      parentChainInfo.blockExplorer,
+      parentChainInfo.chainId,
+      parentChainInfo.name
+    );
+    await checkAddresses(
+      childAddressesToCheck,
+      chain.rpcUrl,
+      chain.explorerUrl,
+      chain.chainId,
+      chain.name
+    );
+  });
+
+export const orbitChainsListSchema = z.object({
+  mainnet: z.record(z.string(), chainSchema),
+  testnet: z.record(z.string(), chainSchema),
+});
+
+// Schema for incoming data from GitHub issue
+export const incomingChainDataSchema = z.object({
+  chainId: z.string().regex(/^\d+$/),
+  name: z.string().min(1),
+  description: z.string().max(250),
+  chainLogo: z.string().url(),
+  color: z.string().regex(/^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/),
+  rpcUrl: z.string().url(),
+  explorerUrl: z.string().url(),
```

Comment:
Are we also making sure that `explorerUrl` doesn't have `/` at the end? Otherwise metamask will throw an error when adding the network